### DISCUSSION TITLE
Does Java Client API uploadSnowballObjects supports auto extraction?

### DISCUSSION THREAD

**Discussion by uni-hungching-chang** [minio/minio#19423] [Q&A] [answered]
Hi, I'm using Minio Java API `uploadSnowballObjects()` and I wonder if it supports auto extraction after uploaded to S3? I've tried including `snowball-auto-extract=true` into userMetadata or headers. After the API call, I observed a .tar file was successfully uploaded under bucketName/ but the .tar file was not automatically extracted. Any suggestions or feedbacks are appreciated!

First attempt: add `snowball-auto-extract` in `userMetadata`
```
            ObjectWriteResponse response = minioClient.uploadSnowballObjects(
                    UploadSnowballObjectsArgs.builder()
                            .bucket(bucketName)
                            .objects(objects)
                            .userMetadata(Map.of(
                                    "snowball-auto-extract", "true"))
                            .build());
```

Second attempt: add `snowball-auto-extract` in `headers`
```
            ObjectWriteResponse response = minioClient.uploadSnowballObjects(
                    UploadSnowballObjectsArgs.builder()
                            .bucket(bucketName)
                            .objects(objects)
                            .headers(Map.of(
                                    "snowball-auto-extract", "true"))
                            .build());
```

A file named `snowball.-[SOME_NUMBER].tar` was created under bucketName/
The Metadata of the uploaded file showed Key: `x-amz-meta-snowball-auto-extract`, Value: `true,true`

<img width="1043" alt="Screen Shot 2024-04-05 at 5 14 50 PM" src="https://github.com/minio/minio/assets/145494207/b2b94191-8218-492d-8ec0-11a1147de855">

Java Minio client version: 8.5.4

**Comment by balamurugana** [comment, marked as answer]
`headers.put("X-Amz-Meta-Snowball-Auto-Extract", "true");` is set by default. You don't need to set it manually.
